src/main/java/com/shopify/model/ShopifyProductUpdateRequest.java

```diff
@@ -148,57 +148,55 @@ public PublishedStep withVariantRequests(final List<ShopifyVariantRequest> varia
 				changed = true;
 			}
 
-			final List<ShopifyVariant> shopifyVariants = new ArrayList<>(variantRequests.size());
-			final List<Integer> positions = new ArrayList<>(variantRequests.size());
+			List<ShopifyVariant> shopifyVariants = new ArrayList<>(variantRequests.size());
+			List<Integer> positions = new ArrayList<>(variantRequests.size());
 
 			for (int i = 0; i < variantRequests.size(); i++) {
 				final ShopifyVariantRequest shopifyVariantRequestForPosition = variantRequests.get(i);
 				positions.add(shopifyVariantRequestForPosition.getRequest().getPosition());
-
 			}
 
-			int maxPosition = variantRequests.stream().map(ShopifyVariantRequest::getRequest)
-					.map(ShopifyVariant::getPosition).max(Comparator.naturalOrder()).get();
+			int maxPosition = positions.stream().max(Comparator.naturalOrder()).get();
+
 			Collections.sort(variantRequests, new ShopifyVariantRequestOption1Comparator());
+
+			updateVariants(variantRequests, shopifyVariants, positions, maxPosition);
+
+			shopifyProduct.setVariants(shopifyVariants);
+
+			return this;
+		}
+
+		private void updateVariants(final List<ShopifyVariantRequest> variantRequests,
```

Comment:
This is a good example of why you shouldn't modify method arguments. You are changing the shopifyVariants in the updateVariants call. This isn't immediately obvious until you inspect that method. It would be better to have this updateVariants be the list that does everything it's already doing but return a list of ShopifyVariants that is then used in the setVariants call. This keeps all values easy to see where they came from and there isn't mystery to whether or not something outside this function changes values of a variable.

Also once you do this, you can remove the shopifyVariants variable from above and just have it assigned right on the updateVariants function. Something like this:

```
final List<ShopifyVariant> shopifyVariants = updateVariants(variantRequests,  positions, maxPosition);
shopifyProduct.setVariants(shopifyVariants);
```